test/skate_web/controllers/redirect_controller_test.exs

```diff
@@ -0,0 +1,49 @@
+defmodule SkateWeb.RedirectTest do
+  use ExUnit.Case, async: true
+
+  alias SkateWeb.Redirect
+
+  defmodule Router do
+    use Phoenix.Router
+
+    get "/1", Redirect, external: "https://test-site-1"
+    get "/1/2", Redirect, external: "https://test-site-2"
+    get "/exceptional", Redirect, []
+  end
+
+  test "an exception is raised when `external` isn't defined" do
+    assert_raise Plug.Conn.WrapperError, ~R[Missing required external: option in redirect], fn ->
+      call(Router, :get, "/exceptional")
+    end
+  end
+
+  test "ignores query parameters" do
+    # Test that a route without a `:external` parameter doesn't pull `external` from query parameter
+    assert_raise Plug.Conn.WrapperError, ~R[Missing required external: option in redirect], fn ->
+      call(Router, :get, "/exceptional?external=https://bad-url")
+    end
+
+    # Test that a route with a `:external` parameter doesn't redirect to a `external` query parameter
+    conn = call(Router, :get, "/1?external=https://bad-url")
+
+    assert Phoenix.ConnTest.redirected_to(conn) == "https://test-site-1"
+  end
```

Comment:
I may be overly cautious here trying to avoid a [CWE-601](https://cwe.mitre.org/data/definitions/601.html). I don't think I know enough about when phoenix calls a plug with `opts` from `init` and when the `opts` contain `params` and what those `params` contain to be able to say if this test is needed or not.

Reply:
It doesn't cover all the ways that a attacker could try and sneak a redirect URL into the request (e.g., headers), but I figured this was one of the easy tests to ensure expected functionality.